### ISSUE TITLE
constructor-super throws a TypeError if a constructor has multiple return statements

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#5916]
**What version of ESLint are you using?** 2.8.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```json
{
  "rules": {
    "constructor-super": 2
  },
  "env": {
    "es6": true
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

I ran eslint on this file:

```js
// test.js
var Foo = class {};
var Baz = class extends Foo {
  constructor () {
    super();
    return 'anything';
    return 'anything';
  }
};
```

**What did you expect to happen?**

I expected ESLint to exit silently, or perhaps give a warning message about the double return statement.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint threw a TypeError:

```
TypeError: Cannot set property 'calledInEveryPaths' of undefined
    at EventEmitter.ReturnStatement (path/to/node_modules/eslint/lib/rules/constructor-super.js:351:70)
    at emitOne (events.js:90:13)
    at EventEmitter.emit (events.js:182:7)
    at NodeEventGenerator.enterNode (path/to/node_modules/eslint/lib/util/node-event-generator.js:42:22)
    at CodePathAnalyzer.enterNode (path/to/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:609:23)
    at CommentEventGenerator.enterNode (path/to/node_modules/eslint/lib/util/comment-event-generator.js:99:23)
    at Controller.traverser.traverse.enter (path/to/node_modules/eslint/lib/eslint.js:887:36)
    at Controller.__execute (path/to/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (path/to/node_modules/estraverse/estraverse.js:501:28)
    at Controller.Traverser.controller.traverse (path/to/node_modules/eslint/lib/util/traverser.js:38:33)
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I have fixed this in #5897.
Please wait for the next release :wink:
